nixos/modules/services/web-apps/lotide.nix

```diff
@@ -0,0 +1,104 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.lotide;
+in
+{
+  options.services.lotide = {
+    enable = mkEnableOption "the lotide service";
+
+    package = mkOption {
+      default = pkgs.lotide;
+      defaultText = "pkgs.lotide";
+      type = types.package;
+      description = "lotide package to use.";
+    };
+
+    databaseUrl = mkOption {
+      default = null;
+      type = types.string;
+      description = "Credentials for database connection.";
+      example = "postgresql://user:passwort@somehost:5432/databasename";
+    };
+
+    hostUrlActivityPub = mkOption {
+      default = null;
+      type = types.string;
+      description = "If using the recommended proxy setup, set this to your root address";
+      example = "https://example.com";
+    };
+
+    hostUrlAPI = mkOption {
+      default = null;
+      type = types.string;
+      description = "URL for API";
+
+      example = "https://example.com/api";
+    };
+
+    apubProxyRewrites= mkOption {
+      default = false;
+      type = types.bool;
+      description = "Set to `true` to make signatures work with the proxy setup.";
+    };
+
+    allowForwarded = mkOption {
+      default = null;
+      type = types.string;
+      description = "Set to true to make ratelimiting work with the proxy setup.";
+    };
+
+    smtpUrl = mkOption {
+      default = null;
+      type = types.string;
+      description = "URL used to access SMTP server, required for sending email";
+
+      example = "smtps://username:password@smtp.example.com";
+    };
+
+    smtpFrom = mkOption {
+      default = null;
+      type = types.string;
+      description = "From value used in sent emails, required for sending email";
+    };
+
+    mediaLocation = mkOption {
+      default = null;
+      type = types.string;
+      description = ''
+        Directory on disk used for storing uploaded images. If not set, image uploads will be disabled.
+        Must be created before this service is started.
+      '';
+
+      example = "/var/lotide/imgs";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.lotide = {
+      description = "lotide Service";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "networking.target" ];
+      serviceConfig = {
```

Comment:
Maybe you should consider adding `DynamicUser = true;` here. We probably don't want this service to run as the `root` user.